src/gui/tray/Window.qml

```diff
@@ -725,34 +725,29 @@ Window {
                                 anchors.centerIn: parent
 
                                 // transform model to contain indexed actions with primary action filtered out
-                                function actionListToContextMenuList(actionList, skipTransform) {
+                                function actionListToContextMenuList(actionList) {
                                     // early out with non-altered data
-                                    if (skipTransform === true) {
+                                    if (activityItem.links.length <= activityListView.maxActionButtons) {
                                         return actionList;
                                     }
 
-                                    // add index to every action
-                                    var actionsWithIndices = actionList.map(function(action, index) {
+                                    // add index to every action and filter 'primary' action out
+                                    var reducedActionList = actionList.reduce(function(reduced, action, index) {
+                                        if (!action.primary) {
+                                           var actionWithIndex = { actionIndex: index, label: action.label };
+                                           reduced.push(actionWithIndex);
+                                        }
+                                        return reduced;
+                                    }, []);
 
-                                        var actionWithIndex = { actionIndex: index, label: action.label, primary: action.primary };
 
-                                        return actionWithIndex;
-                                    });
-
-                                    // filter 'primary' action out
-                                    var filteredActions = actionsWithIndices.filter(function(action) {
-                                        return action.primary !== true;
-                                    });
-
-                                    return filteredActions;
+                                    return reducedActionList;
                                 }
 
                                 Repeater {
                                     id: moreActionsButtonContextMenuRepeater
 
-                                    readonly property bool skipTransformModel: activityItem.links.length <= activityListView.maxActionButtons
-
-                                    model: moreActionsButtonContextMenu.actionListToContextMenuList(activityItem.links, skipTransformModel)
+                                    model: moreActionsButtonContextMenu.actionListToContextMenuList(activityItem.links)

```

Comment:
Can't it be `parent.actionListToContextMenuList(activityItem.links)` now?

Reply:
You are really following the path of perfection @er-vin  👍